crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -76,12 +111,88 @@ impl Settlement {
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
-    #[error(transparent)]
-    Solution(#[from] solution::Error),
+    #[error("failed communication with the database: {0}")]
+    BadCommunication(sqlx::Error),
```

Comment:
naming nit (bad communication when handles higher up in the domain is not really descriptive IMO, we should make it clear that it's an infra or persistence error).

```suggestion
    Infra(sqlx::Error),
```

Reply:
renamed